src/parser/LiteralOrIriType.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023, University of Freiburg,
+//                 Chair of Algorithms and Data Structures.
+// Author: Benedikt Maria Beckermann <benedikt.beckermann@dagstuhl.de>
+
+#include "LiteralOrIriType.h"
+
+LiteralOrIriType::LiteralOrIriType(IriType data) : data(data) {}
+LiteralOrIriType::LiteralOrIriType(LiteralType data) : data(data) {}
+
+bool LiteralOrIriType::isIri() const {
+  return std::holds_alternative<IriType>(data);
+}
+
+IriType& LiteralOrIriType::getIriTypeObject() {
+  if (!isIri()) {
+    AD_THROW(
+        "LiteralOrIriType object does not contain an IriType object and thus "
+        "cannot return it");
+  }
```

Comment:
Also use `AD_CONTRACT_CHECK as soon as I've merged it with the additional message.